packager/media/formats/mp2t/ts_writer.cc

```diff
@@ -188,14 +197,40 @@ void TsWriter::SignalEncrypted() {
 }
 
 bool TsWriter::AddPesPacket(std::unique_ptr<PesPacket> pes_packet,
-		            BufferWriter* buffer) {
+                            BufferWriter* buffer) {
+  if (!WritePesToBuffer(*pes_packet, &elementary_stream_continuity_counter_,
+                        buffer)) {
+    LOG(ERROR) << "Failed to write pes to buffer.";
+    return false;
+  }
 
+  // No need to keep pes_packet around so not passing it anywhere.
+  return true;
+}
+
+bool TsStuffingWriter::AddPesPacket(std::unique_ptr<PesPacket> pes_packet,
+                                    BufferWriter* buffer) {
   if (!WritePesToBuffer(*pes_packet, &elementary_stream_continuity_counter_,
                         buffer)) {
     LOG(ERROR) << "Failed to write pes to buffer.";
     return false;
   }
 
+  // We must end all ES packets at 0xf so that the next segment can start at
+  // 0x0. This can be done by stuffing null packets at the end of the segment
+  // for each elementary stream
+  do {
```

Comment:
As discussed, I think it would be cleaner to move this code block to `TsSegmenter::FinalizeSegment()`.  We could add the option for CC stuffing in `MuxerOptions` to toggle this behavior.  WDYT?